hazel.bzl

```diff
@@ -167,20 +172,19 @@ def hazel_custom_package_github(
     strip_prefix: strip this path prefix from directory repo, useful when a
                   repo contains several packages.
     archive_sha256: hash of the actual archive to download.
+    clone_via_ssh: whether to clone the repo using SSH (useful for private
+                   repos).
   """
-  url = "https://github.com/{0}/{1}/archive/{2}.tar.gz".format(
-    github_user,
-    github_repo,
-    repo_sha,
-  )
+
   fixed_package_name = _fixup_package_name(package_name)
-  strip_prefix_combined = "{0}-{1}".format(github_repo, repo_sha)
-  if strip_prefix:
-    strip_prefix_combined += "/" + strip_prefix
-  native.new_http_archive(
+  build_file = "//third_party/haskell:BUILD.{0}".format(fixed_package_name)
+  url = "https://github.com/{0}/{1}".format(github_user, github_repo)
```

Comment:
I suppose eventually we may want to support more sources than just GitHub; at that point, it may make sense to just take a full URL rather than reconstructing it.  But given that we currently have to write a new rule for each subdirectory, this seems reasonable for now.